### PR TITLE
Persistence WIP - against develop

### PR DESCRIPTION
**Pull request by mcovarr**
Either this or #64 should be closeable.

### CODE REVIEW

File: src/main/scala/cromwell/engine/CallActor.scala [hunk 1]
```diff
@@ -51,15 +48,18 @@ class CallActor(call: Call, backend: Backend, workflowDescriptor: WorkflowDescri
...
     val originalSender = sender()
+    val backendInputs = backend.adjustInputPaths(call, locallyQualifiedInputs)
+    log.info("Got backend inputs for " + call.fullyQualifiedName + ": " + backendInputs)
 
-    for {
-      inputs <- (storeActor ? StoreActor.GetLocallyQualifiedInputs(call)).mapTo[Map[String, WdlValue]]
-      backendInputs = backend.adjustInputPaths(call, inputs)
-      commandLine <- Future.fromTry(call.instantiateCommandLine(backendInputs))
+    val tryCommand = for {
+      commandLine <- call.instantiateCommandLine(backendInputs)
```
Comment (geoffjentry) [inline src/main/scala/cromwell/engine/CallActor.scala]: See the comment I made in #67

File: src/main/scala/cromwell/engine/workflow/WorkflowActor.scala [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+  when(WorkflowRunning) {
+    case Event(CallStarted(call), NoFailureMessage) =>
+      persistStatus(call, Running)
+      stay()
+    case Event(CallCompleted(call, outputs), NoFailureMessage) =>
+      // See DSDEEPB-521 for an example of why potential race conditions in starting runnable calls currently
+      // require this. Could Cromwell be smarter about the way it looks for runnable calls?
+      Await.result(handleCallCompleted(call, outputs), DatabaseTimeout)
+      if (isWorkflowDone) {
+        goto(WorkflowSucceeded)
```
Comment (geoffjentry) [inline src/main/scala/cromwell/engine/workflow/WorkflowActor.scala]: https://www.cs.utexas.edu/users/EWD/ewd02xx/EWD215.PDF

I kid, I kid

File: src/main/scala/cromwell/engine/workflow/WorkflowActor.scala [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+        _ <- persistStatus(runnableCalls, Starting)
+        allInputs <- Future.sequence(runnableCalls map fetchLocallyQualifiedInputs)
+      } yield runnableCalls zip allInputs
+      val callsAndInputs = Await.result(futureCallsAndInputs, DatabaseTimeout)
+      log.info("Calls and inputs are " + callsAndInputs)
+      callsAndInputs foreach { case (call, inputs) => startActor(call, inputs)}
+    }
+  }
+
+  // FIXME FIXME FIXME this needs to be refactored but I don't understand it.
```
Comment (geoffjentry) [inline src/main/scala/cromwell/engine/workflow/WorkflowActor.scala]: This function is a bag of hurt. I'm surprised that Odersky hasn't taken away our scala license. Perhaps I've found something to do in the car on my trip? :)

File: src/main/scala/cromwell/engine/workflow/WorkflowActor.scala [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+
+  private def fetchCallOutputEntries(call: Call) = dataAccess.getOutputs(workflow.id, call)
+  private def fetchCallInputEntries(call: Call) = dataAccess.getInputs(workflow.id, call)
+
+  /**
+   * Load whatever execution statuses are stored for this workflow, regardless of whether this is a workflow being
+   * restarted, or started for the first time.
+   */
+  private def createStore: Future[Map[Call, ExecutionStatus.Value]] = {
+    for {
```
Comment (geoffjentry) [inline src/main/scala/cromwell/engine/workflow/WorkflowActor.scala]: I don't understand the for/yield w/ a single clause. Is it getting you something?

File: src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+      }
+    }
+
+    type QuantifierAndPlural = (String, String)
+    def pluralize(num: Int): QuantifierAndPlural = {
+      (if (num == 0) "no" else num.toString, if (num == 1) "" else "s")
+    }
+
+    val result = for {
+      workflowInfos <- dataAccess.getWorkflowsByState(Seq(WorkflowSubmitted, WorkflowRunning))
```
Comment (geoffjentry) [inline src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala]: Similar comment on the single clause for comprehension. Also the nested for comprehensions makes me a little twitchy.

File: src/main/scala/cromwell/engine/workflow/WorkflowActor.scala (hunk 3 above)
Comment (mcovarr) [inline src/main/scala/cromwell/engine/workflow/WorkflowActor.scala]: Even just documenting and/or decomposing into functions would be a win here IMHO.

File: src/main/scala/cromwell/engine/workflow/WorkflowActor.scala (hunk 4 above)
Comment (mcovarr) [inline src/main/scala/cromwell/engine/workflow/WorkflowActor.scala]: This is vestigial, there used to be more stuff here.  I'll update this to a simple `map`.

File: src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala (hunk 5 above)
Comment (mcovarr) [inline src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala]: I'd agree with you here too except I've added more stuff on yet-uncommitted code to do restart properly, so I'd like to leave this as is to minimize imminent diffs.  :smile:

Comment (geoffjentry) [inline src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala]: +1

**Pull request merged**